blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/NodeRunStatus.java

```diff
@@ -0,0 +1,84 @@
+package io.jenkins.blueocean.rest.impl.pipeline;
+
+import io.jenkins.blueocean.rest.model.BlueRun;
+import org.jenkinsci.plugins.workflow.actions.NotExecutedNodeAction;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.pipelinegraphanalysis.GenericStatus;
+
+/**
+ * @author Vivek Pandey
+ */
+public class NodeRunStatus {
+    public final BlueRun.BlueRunResult result;
+    public final BlueRun.BlueRunState state;
+
+    public NodeRunStatus(FlowNode endNode) {
```

Comment:
Ignore this, NodeRunStatus(FlowNode endNode) is for legacy graph builder, its disabled by default and will be removed eventually soon.